<html>

<head>
<title>NodeBox | crash and python location</title>
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta http-equiv="imagetoolbar" content="no" />
<meta http-equiv="linode" content="true"/>
<meta name="description" content="" />
<meta name="keywords" content="NodeBox, shared 2007-02-12-12-00-32, bug" />
<link rel="alternate" type="application/rss+xml" title="NodeBox News Feed" href="https://nodebox.net/code/data/rss.xml" />
<link rel="canonical" href="https://nodebox.net/code/index.php/shared_2007-02-12-12-00-32" />
<link type="text/css" rel="stylesheet" media="screen" href="../../media/css/nbar.css">

<script type="text/javascript" src="../js/pop.js"></script>
<script type="text/javascript" src="../js/confirm.js"></script>
<link href="../css/default.css" rel="stylesheet" type="text/css" />
<link href="../css/print.css" rel="stylesheet" type="text/css" media="print" />
<!--[if gte IE 5.5]>
<script type="text/javascript" src="/code/js/pngfix.js"></script>
<![endif]-->

<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-1668598-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>
</head>

<body id="body" >
<div class="nbar">
  <ol>
    <li><a class="node" href="https://www.nodebox.net/code/">NodeBox 1<span class="nbar-arrow"></span></a>
      <ol class="nbar-dropdown">
        <li><a href="https://www.nodebox.net/">Homepage</a></li>
        <li><a href="https://www.nodebox.net/node/">NodeBox 3<small>Node-based app for generative design and data visualization</small></a></li>
        <li><a href="https://www.nodebox.net/opengl/">NodeBox OpenGL<small>Hardware-accelerated cross-platform graphics library</small></a></li>
        <li><a href="https://www.nodebox.net/code/">NodeBox 1<small>Generate 2D visuals using Python code (Mac OS X only)</small></a></li>
      </ol>
    </li>
    <li><a class="gallery" href="https://www.nodebox.net/gallery/">Gallery</a></li>
    <li><a class="documentation" href="Tutorial.html">Documentation</a></li>
    <li><a class="forum" href="http://support.nodebox.net/discussions">Forum</a></li>
    <li><a class="blog" href="https://www.nodebox.net/blog/">Blog</a></li>
  </ol>
</div>


<div id="all">
<div id="header_and_navigation">

<div id="header">
<a href="Home.html"><img id="header_image" src="../g/header-small.jpg" width="800" /></a></div>

<div id="title">
<a href="Home.html"><h1>NodeBox</h1>
<strong><em>Create visual output with Python programming code</em></strong>
</a></div>

<div id="contextual">
<div id="languages">
 
</div>

<div id="search">
<form id = "sf" method="get">
<input type="text" id="q" name="q" value="" />
<a href="javascript:document.getElementById('sf').submit();">GO</a>
</form>
</div>

</div>

<div id="navigation">
<div id="navigation_public">
<a id="" href="Home.html">Home</a> 
<a id="" href="Download.html">Download</a> 
<a id="" href="Reference.html">Reference</a> 
<a id="" href="Tutorial.html">Tutorial</a> 
<a id="" href="Library.html">Library</a> 
<a id="" href="Gallery.html">Gallery</a> 
<a id="" href="Share.html">Share</a> 
<a id="" href="About.html">About</a> 
</div>
</div>
 
</div>

<div id="content">

<div id="nodebox-link">
<a href="Home.html"><img src="../g/transparent.gif" /></a>
</div>

<div class="inside">
<? if (do_name() != 'Home') : ?>
 <? endif ?>
<br /><h1 id="shared-title">crash and python location</h1> 			<strong>Posted by <span class="comment_by">todd</span> on <em class="date">Feb 12, 2007</em></strong> 			<p> 			 Hi,<br /> When first starting nodebox i received an error that it could not find the python runtime and to change this in the Info.plist.<br /> <br /> I did this as I only have the 2.4 Framework version installed which now lives in <br /> /Library/Frameworks/Python.framework/Versions/2.4/Python<br /> <br /> Now it crashes on startup. Below is the crash report. Do I need to have 2.3 installed? I did not see that as a requirement.<br /> <br /> <br /> <br /> Date/Time:      2007-02-12 08:23:02.843 -0700<br /> OS Version:     10.4.8 (Build 8L127)<br /> Report Version: 4<br /> <br /> Command: NodeBox<br /> Path:    /Applications/NodeBox/NodeBox.app/Contents/MacOS/NodeBox<br /> Parent:  WindowServer [11421]<br /> <br /> Version: 1.8.3 ()<br /> <br /> PID:    1271<br /> Thread: 0<br /> <br /> Exception:  EXC_BAD_ACCESS (0x0001)<br /> Codes:      KERN_PROTECTION_FAILURE (0x0002) at 0x00000000<br /> <br /> Thread 0 Crashed:<br /> 0   com.apple.CoreFoundation 	0x907bb574 CFStringGetCString + 84<br /> 1   net.nodebox.NodeBox      	0x00006af8 getPythonInterpreter + 15256<br /> 2   net.nodebox.NodeBox      	0x00007960 main + 1920<br /> 3   net.nodebox.NodeBox      	0x000017f4 start + 404<br /> 4   net.nodebox.NodeBox      	0x0000169c start + 60<br /> <br /> Thread 0 crashed with PPC Thread State 64:<br />   srr0: 0x00000000907bb574 srr1: 0x000000000200f030                        vrsave: 0x0000000000000000<br />     cr: 0x22004244          xer: 0x0000000000000007   lr: 0x00000000907bb530  ctr: 0x00000000907bb520<br />     r0: 0x00000000a07ba674   r1: 0x00000000bfffe9e0   r2: 0x00000000a07ba274   r3: 0x0000000000000000<br />     r4: 0x00000000bfffef5c   r5: 0x0000000000000400   r6: 0x0000000008000100   r7: 0x0000000000000400<br />     r8: 0x0000000000000001   r9: 0x000000000000001c  r10: 0x0000000000000000  r11: 0x0000000044000444<br />    r12: 0x00000000907bb520  r13: 0x00000000000062fc  r14: 0x00000000bfffef5c  r15: 0x0000000000312910<br />    r16: 0x00000000bffffb70  r17: 0x00000000bfffeafc  r18: 0x00000000000062fc  r19: 0x00000000000062fc<br />    r20: 0x0000000000000002  r21: 0x0000000000301c80  r22: 0x00000000000062fc  r23: 0x00000000bffff75c<br />    r24: 0x00000000bfffef5c  r25: 0x0000000000000400  r26: 0x0000000000301c90  r27: 0x0000000000000000<br />    r28: 0x000000000031261d  r29: 0x0000000008000100  r30: 0x00000000bfffea60  r31: 0x00000000907bb530<br /> <br /> Binary Images Description:<br />     0x1000 -     0x8fff net.nodebox.NodeBox 1.8.3 ()	/Applications/NodeBox/NodeBox.app/Contents/MacOS/NodeBox<br /> 0x8fe00000 - 0x8fe51fff dyld 45.3	/usr/lib/dyld<br /> 0x90000000 - 0x901bcfff libSystem.B.dylib 	/usr/lib/libSystem.B.dylib<br /> 0x90214000 - 0x90219fff libmathCommon.A.dylib 	/usr/lib/system/libmathCommon.A.dylib<br /> 0x907ba000 - 0x90893fff com.apple.CoreFoundation 6.4.6 (368.27)	/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation<br /> 0x908de000 - 0x909e0fff libicucore.A.dylib 	/usr/lib/libicucore.A.dylib<br /> 0x90a3a000 - 0x90abefff libobjc.A.dylib 	/usr/lib/libobjc.A.dylib<br /> 0x90b6e000 - 0x90b80fff libauto.dylib 	/usr/lib/libauto.dylib<br /> 0x94beb000 - 0x94c0bfff libmx.A.dylib 	/usr/lib/libmx.A.dylib<br /> <br /> Model: PowerBook5,2, BootROM 4.7.1f1, 1 processors, PowerPC G4  (1.1), 1 GHz, 1 GB<br /> Graphics: ATI Mobility Radeon 9600, ATY,RV350M10, AGP, 64 MB<br /> Memory Module: SODIMM1/J25UPPER, 1 GB, DDR SDRAM, PC2700U-25330<br /> AirPort: AirPort Extreme, 405.1 (3.90.34.0.p18)<br /> Modem: LastDash, UCJ, V.92, 4.0, APPLE VERSION 2.6.6<br /> Bluetooth: Version 1.7.8f2, 2 service, 0 devices, 1 incoming serial ports<br /> Network Service: AirPort, AirPort, en1<br /> PCI Card: TXN,PCIXXXX-00, cardbus, PC Card<br /> Parallel ATA Device: MATSHITACD-RW  CW-8123, <br /> Parallel ATA Device: Hitachi IC25N060ATMR04-0, 55.89 GB<br /> USB Device: Bluetooth HCI, , Up to 12 Mb/sec, 500 mA<br />  			</p> 			<br /> 			<div class="comment"><hr /><strong>Posted by <span class="comment_by">Frederik De Bleser</span> on <em class="date">Feb 13, 2007</em></strong><br /><p> NodeBox is compiled using the --semi-standalone flag, which means it expects a Python to be already installed. Normally, this is no problem, since on Mac OS X 10.4, Python 2.3.5 is installed by default in /System/Library/Frameworks/Python/Version/2.3/.<br /> <br /> It&#39;s kind of strange that the default Python wouldn&#39;t work. It could be that you permissions are set wrong. Can you run python2.3 from the Terminal?<br /> <br /> Kind regards,<br /> <br /> Frederik</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">todd</span> on <em class="date">Feb 13, 2007</em></strong><br /><p> Actually I do not have 2.3 on my system as I removed it. I replace it with 2.4. I will fix this and see if it works.</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">todd</span> on <em class="date">Feb 13, 2007</em></strong><br /><p> ok. it is all fixed now. sorry for the confusion.</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Chris</span> on <em class="date">Feb 14, 2007</em></strong><br /><p> Hi, I thought that it was in:<br /> /System/Library/Frameworks/Python.framework/Version/2.3/<br /> <br /> Mine is there and I get the same crash....</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Frederik De Bleser</span> on <em class="date">Feb 14, 2007</em></strong><br /><p> Is there something in the Console window (/Applications/Utilities/Console)? Clear it out, then run NodeBox to see if anythin appears there. Otherwise, could you mail me the crash log? My e-mail address if frederik at the domain name pandora.be</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Frederik De Bleser</span> on <em class="date">Feb 15, 2007</em></strong><br /><p> I found out that the problem (again) is input managers. You might want to read a <a href="http://nodebox.net/code/index.php/shared_2007-01-17-12-00-23" target="_self">previous form post</a> on this.</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Neal</span> on <em class="date">Feb 26, 2007</em></strong><br /><p> I had removed python2.3 from my system, so I reinstalled it using an old macpython installer.  It went in /Library/Frameworks/... instead of /System/Library/Frameworks/..<br /> <br /> This seems to work fine for most of nodebox, but when I tried using the Graph library, I get an error <br /> <br /> </p><pre class="python">Traceback <span style="">&#40;</span>most recent call last<span style="">&#41;</span>:
   File <span style="color: #ff0080;">&quot;/Users/Shared/Applications/NodeBox/NodeBox.app/Contents/Resources/NodeBox.py&quot;</span>, <span style="color: #530035;">line</span> <span style="">337</span>, <span style="color: #530035;">in</span> _execScript
   File <span style="color: #ff0080;">&quot;/Users/Shared/Applications/NodeBox/Examples/downloads/graph-ppc/graph_test.py&quot;</span>, <span style="color: #530035;">line</span> <span style="">6</span>, <span style="color: #530035;">in</span> ?
   File <span style="color: #ff0080;">&quot;DrawingPrimitives.pyc&quot;</span>, <span style="color: #530035;">line</span> <span style="">1039</span>, <span style="color: #530035;">in</span> <span style="color: #530035;">ximport</span>
   File <span style="color: #ff0080;">&quot;/Users/Shared/Applications/NodeBox/Examples/downloads/graph-ppc/graph.py&quot;</span>, <span style="color: #530035;">line</span> <span style="">5</span>, <span style="color: #530035;">in</span> ?
   File <span style="color: #ff0080;">&quot;/Users/tom/Desktop/bgl-python-0.9/python/boost/graph/__init__.py&quot;</span>, <span style="color: #530035;">line</span> <span style="">9</span>, <span style="color: #530035;">in</span> ?
 <span style="">ImportError</span>: Failure linking <span style="">new</span> module: /Users/Shared/Applications/NodeBox/Examples/downloads/graph-ppc/boost/graph/_support.<span style="">so</span>: Symbol <span style="color: #530035;">not</span> found: _PyUnicodeUCS2_AsWideChar
   Referenced <span style="color: #530035;">from</span>: libboost_python.<span style="">dylib</span>
   Expected <span style="color: #530035;">in</span>: /Library/Frameworks/Python.<span style="">framework</span>/Versions/<span style="">2.3</span>/Python</pre></pre><p>&nbsp;</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Frederik De Bleser</span> on <em class="date">Feb 27, 2007</em></strong><br /><p> Hi Neal,<br /> <br /> You might want to try building boost from the source. I included instructions on our developers wiki: <a href="http://dev.nodebox.net/wiki/BuildingBoost">http://dev.nodebox.net/wiki/BuildingBoost</a><br /> <br /> Kind regards,<br /> <br /> Frederik</p><br /></div><div class="comment"><hr /><strong>Posted by <span class="comment_by">Neal</span> on <em class="date">Feb 27, 2007</em></strong><br /><p> <br /> Hi, thanks.  It turns out it was a problem with the python installation, in particular with the permissions.  By copying an existing python 2.3.5 framework from another mac, and being careful to get the permissions exactly right, I fixed the problem.<br /> <br /> </p><br /></div> 	<? include("util/comment.php"); ?>
</div>
  
</div>

<div id="footer">
Last modified: Apr 11 2008 | &copy; 2004-2012 <a href="http://www.emrg.be/" class="noexternal">Experimental Media Research Group</a>
</div>
<img src="../g/footer.jpg" style="border:0" class="footer" width="800" height="334" />

</div>

<script>
try{e=document.getElementsByTagName("span");for(i=0;i<e.length;i++){if(e[i].className=="header_image"){src=e[i].getElementsByTagName("img")[0].src;document.getElementById("header_image").src=src;break;}}}catch(e){}
</script>
</body>

</html>